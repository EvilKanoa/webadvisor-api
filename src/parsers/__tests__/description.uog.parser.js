const path = require('path');
const util = require('util');
const readFile = util.promisify(require('fs').readFile);
const { parseDescription } = require('../description.uog.parser');

describe('UOG Course Description Parser', () => {
  it('handles empty inputs', async () => {
    expect(await parseDescription('')).toBeFalsy();
    expect(await parseDescription(null)).toBeFalsy();
    expect(await parseDescription(false)).toBeFalsy();
    expect(await parseDescription('<html></html>')).toBeFalsy();
  });

  it.each([
    [
      'parses a description correctly',
      'description.uog.data.1.html',
      'CIS*2750',
    ],
    [
      'fails gracefully when course not found',
      'description.uog.data.2.html',
      'CIS*2750',
    ],
  ])('%s', async (desc, filename, code) => {
    const html = await readFile(path.join(__dirname, '__data__', filename));
    const result = await parseDescription(html, code);
    const str = util.inspect(result, {
      depth: 10,
      maxArrayLength: Infinity,
      sorted: true,
    });
    expect(str).toMatchSnapshot();
  });
});
