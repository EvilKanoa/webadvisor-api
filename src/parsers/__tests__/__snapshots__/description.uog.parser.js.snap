// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description.uog.parser fails gracefully when course not found 1`] = `"''"`;

exports[`description.uog.parser parses a description correctly 1`] = `"'This course introduces techniques and tools used in the development of large software systems. Students learn methods for organizing and constructing modular systems, manipulating files, introductory interface design, and use of databases. Software tools for managing projects, database connectivity, configuration management, and system application programmer interfaces are also covered.'"`;
